### ISSUE TITLE
[BUG] EFB crash on landing perf screen with input locked

### ISSUE THREAD

**Issue by timdawg** [invernyx/md11-bugs#1221]
### Describe the bug

EFB crash on landing perf screen with input locked

### Steps to reproduce

Used the EFB for landing perf calculator. Returned later to update the landing weight and the airport field was empty.  When attempting to type in the airport ICAO, the EFB went black with just this error message in the top left: **undefined is not an object (evaluating 'runway slope')**

All controls mapped in MSFS stopped working, because the input was locked to the EFB when it crashed. This leaves the user unable to change camera views or control the plane, unless they have a third party app like Flow or Spad.Next that can manipulate the sim variables and events directly.

### Expected behaviour

Input validation - validate the inputs are valid before attempting calculations to prevent errors from occurring.

Error handling - better error handling to prevent the entire EFB from crashing, like display a dialog and reset the form.

If the EFB crashes for some reason, it should release the input so that it does not prevent the user from continuing their flight.

### Evidence

![2024-05-25 10_53_50-Microsoft Flight Simulator - 1 37 18 0](https://github.com/invernyx/md11-bugs/assets/404459/77423258-cc06-4f2f-ac02-7c199c68a102)

### Simulator

Microsoft Flight Simulator 2020 (MSFS)

### Crash

No

### Agreement

- [X] Yes

**Comment by JohnWilkins82** [comment]
Same as above
![md11EFBCrash](https://github.com/invernyx/md11-bugs/assets/58406371/64e93caa-4e6d-498a-a074-cb708fa8999c)

**Comment by KJA1582** [comment]
As a workaround until this is fixed, open http://127.0.0.1:19999 in your browser.

Find the EFB in the list (named TFDi_MD11_efb). Click that, and once it loaded, hit the reload icon (see picture).
That'll reload the EFB.
You can then click on set simbrief username which will navigate to the settings.
Click into the textbox and exit the textbox, controls restored.

![image](https://github.com/invernyx/md11-bugs/assets/2170028/5bd8afa6-504f-4875-b9f7-18b27f2e4bc1)

**Comment by GenericNerd** [comment]
Can you please provide an exact, step-by-step guide to reproduce this issue?

**Comment by KJA1582** [comment]
I'm not sure why it triggers.
The first time I entered my dest. airport (WSSS) it worked as expected, gave me all the data /That was shortly after T/C cause I was curious)

When I tried to do the actual approach prep before T/D (80nmi or so) I reopened the page. The Airport field was Blank. The first typed letter resulted in the `undefined` object access.

Using the Coherent Debug I restarted the EFB so as to not waste an 8h flight. I did not attempt another calculation

**Comment by timdawg** [comment]
I have found 2 methods that reproduce the the issue. I recorded it and attached the videos below.

This only occurs after all the info has been entered for the landing perf calculator and the landing distance has been calculated.

Primary method: Change tabs to another tab, then return to the EFB. The ICAO field will be blank. If you start typing in the ICAO code, the error will occur.
https://github.com/invernyx/md11-bugs/assets/404459/7f5e9ce6-e9bf-4447-9fd7-370ad6052689

Alternate method: Backspace in the ICAO field
https://github.com/invernyx/md11-bugs/assets/404459/37bce795-da17-4d1f-af40-c3accc173f94

I did find that you do not need to re-type the ICAO code, even though it is blank. It will still work and even allow you to change runways, so that is an even better workaround to prevent the issue.  My instinct was to fill that field out again, and that caused the error.

Maintaining the ICAO code after tab changes will reduce the likelihood of the error, however it would still not be possible to calculate for an alternate destination, after running the calculations for the primary.

If you disable the ICAO code field, when the landing distance is been calculated, and show a change button next to it, that resets the entire form, I think that would successfully mitigate the issue.  Or reset the fields causing the issue whenever the ICAO code is changed.
